site/js/components/ee.js

```diff
@@ -0,0 +1,227 @@
+    }
+  }
+
+  /**
+   * Start over
+   * @return {void}
+   */
+  clear() {
+    this._clearQuizStart();
+    this._clearCurrentPrompt();
+  }
+
+  /**
+   * Gets the current prompt value from local storage.
+   * @return {int} - current prompt value
+   */
+  _getCurrentPrompt() {
+    const curPrompt = parseInt(localStorage.getItem('chromeos-m100-current-prompt'));
+    return curPrompt;
+  }
+
+  /**
+   * Sets the current prompt value.
+   * @param {int} value - the current prompt integer value
+   * @return {void}
+   */
+  _setCurrentPrompt(value) {
+    localStorage.setItem('chromeos-m100-current-prompt', value);
+    this.curPrompt = value;
+  }
+
+  /**
+   * Clear the current prompt value.
+   * @return {void}
+   */
+  _clearCurrentPrompt() {
+    localStorage.removeItem('chromeos-m100-current-prompt');
+    this.curPrompt = 0;
+  }
+
+  /**
+   * Gets the current quiz start value from local storage.
+   * @return {int} - current quiz start value
+   */
+  _getQuizStart() {
+    const quizStart = !!localStorage.getItem('chromeos-m100-quiz-start');
+    return quizStart;
+  }
+
+  /**
+   * Sets the current quiz start value.
+   * @param {int} value - the current quiz start integer value
+   * @return {void}
+   */
+  _setQuizStart(value) {
+    localStorage.setItem('chromeos-m100-quiz-start', value);
+    this.quizStart = value;
+  }
+
+  /**
+   * Clear the current quiz start value.
+   * @return {void}
+   */
+  _clearQuizStart() {
+    localStorage.removeItem('chromeos-m100-quiz-start');
+    this.quizStart = false;
+  }
+
+  /**
+   * Prompt the question to the user
+   * @param {int} prompt
+   */
+  _promptQuestion(prompt = 0, ret = false) {
+    if (prompt < this.text.prompts.length && this.text.prompts[prompt].question.length > 0) {
+      if (ret) {
+        return this.text.prompts[prompt].question;
+      }
+      console.info('Question #%s: %c%s', prompt + 1, 'font-weight: bold;', this.text.prompts[prompt].question);
+    } else if ((prompt = this.text.prompts.length)) {
+      this._promptReward();
+    }
+  }
+
+  /**
+   * Prompt the hint to the user
+   * @param {int} prompt
+   */
+  _promptHint(prompt) {
+    if (this.text.prompts[prompt].hint.length > 0) {
+      console.info('Hint: %c%s', 'font-style: italic;', this.text.prompts[prompt].hint);
+    }
+  }
+
+  /**
+   * Prompt reward
+   * @param {int} prompt
+   */
+  _promptReward() {
+    console.info('%c%s', 'font-weight: bold; font-size: 1.25em;', this.text.reward.headline);
+    for (const iterator of this.text.reward.body) {
+      console.info(iterator);
+    }
+  }
+
+  /**
+   * Check the answer
+   * @param {int} prompt
+   * @param {string} value
+   * @return {void}
+   */
+  _checkAnswer(prompt, value) {
+    // console.info('Question #%s: %c%s', prompt + 1, 'font-weight: bold; font-style: italic;', this._promptQuestion(prompt, true));
+    if (value.toString().toLowerCase() == this.text.answers[prompt].toString().toLowerCase()) {
```

Comment:
With moving answers into the prompt itself, you should be able to grab the `this.text.prompts[prompt]` and reuse that here, for the `responseCorrect`, and `responseIncorrect` instead of needing to construct the lookup each time. It'll also let you pass in the already found prompt to `this._promptHint`